src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -12684,157 +12699,136 @@ namespace SurfaceGeometry {
+                            } else {
+                                thisEnclosure.ExtWindowArea += Zone(surf.Zone).ExtWindowArea;
+                                thisEnclosure.TotalSurfArea += Zone(surf.Zone).TotalSurfArea;
+                                Zone(surf.Zone).SolarEnclosureNum = thisSideEnclosureNum;
+                            }
                         } else if (otherSideEnclosureNum == 0) {
                             // This side is assigned, so use that one for both
                             otherSideEnclosureNum = thisSideEnclosureNum;
-                            auto &thisSolEnclosure(DataViewFactorInformation::ZoneSolarInfo(thisSideEnclosureNum));
-                            thisSolEnclosure.ZoneNames.push_back(Surface(surf.ExtBoundCond).ZoneName);
-                            thisSolEnclosure.ZoneNums.push_back(Surface(surf.ExtBoundCond).Zone);
-                            thisSolEnclosure.FloorArea += Zone(Surface(surf.ExtBoundCond).Zone).FloorArea;
-                            thisSolEnclosure.ExtWindowArea += Zone(Surface(surf.ExtBoundCond).Zone).ExtWindowArea;
-                            thisSolEnclosure.TotalSurfArea += Zone(Surface(surf.ExtBoundCond).Zone).TotalSurfArea;
+                            auto &thisEnclosure(Enclosures(thisSideEnclosureNum));
+                            thisEnclosure.ZoneNames.push_back(Surface(surf.ExtBoundCond).ZoneName);
+                            thisEnclosure.ZoneNums.push_back(Surface(surf.ExtBoundCond).Zone);
+                            thisEnclosure.FloorArea += Zone(Surface(surf.ExtBoundCond).Zone).FloorArea;
+                            if (radiantSetup) {
+                                Zone(Surface(surf.ExtBoundCond).Zone).RadiantEnclosureNum = otherSideEnclosureNum;
+                            } else {
+                                thisEnclosure.ExtWindowArea += Zone(Surface(surf.ExtBoundCond).Zone).ExtWindowArea;
+                                thisEnclosure.TotalSurfArea += Zone(Surface(surf.ExtBoundCond).Zone).TotalSurfArea;
+                                Zone(Surface(surf.ExtBoundCond).Zone).SolarEnclosureNum = otherSideEnclosureNum;
+                            }
                         } else if (thisSideEnclosureNum != otherSideEnclosureNum) {
-                            // This should never happen
-                            ErrorsFound = true;
-                            ShowSevereError(RoutineName + ": Solar enclosure grouping error for Surface=\"" + surf.Name + "\"." +
-                                            "This surface enclosure num=" + General::RoundSigDigits(thisSideEnclosureNum) +
-                                            ". Other side enclosure num=" + General::RoundSigDigits(otherSideEnclosureNum));
+                            // If both sides are already assigned to an enclosure, then merge the two enclosures
+                            auto &thisEnclosure(Enclosures(thisSideEnclosureNum));
```

Comment:
The following section merges two enclosures that are later found to be connected, fixing 7843.